src/api/init.ts

```diff
@@ -320,6 +321,39 @@ export async function startApiServer(opts: {
     });
   };
 
+  const forceKill = async () => {
+    logger.info('Force closing API server...');
+    await Promise.allSettled([
```

Comment:
`Promise.allSettled` seems useful here. But, I think in order to prevent race conditions between the `server.close()` function the the `socket.destroy()` function, these need to be performed in a specific order, synchronously, and _then_ the callbacks for those could be used to "complete" a promise, and awaited inside `Promise.allSettled`.

For example, there's a function `waiter()` in `helper.ts` that returns a `Waiter<T>` object useful for these scenarios.

It could be used to do something like:
```ts
const [ioClosePromise, wssClosePromise, serverClosePromise] = [waiter(), waiter(), waiter()];
// Synchronously call `close` on the NodeJS http server which prevents _new_ connections.
// TODO: preserve the error logging in these close callbacks..
io.close(() => ioClosePromise.finish());
wss.close(() => wssClosePromise.finish());
server.close(() => serverClosePromise.finish());

// Destroy all the existing connections.
for (const socket of serverSockets) {
  socket.destroy();
}

await Promise.allSettled([ioClosePromise, wssClosePromise, serverClosePromise]);
```